server/src/helpers/errors.ts

```diff
@@ -101,15 +101,31 @@ const errors: Record<number, RosettaError> = {
 
 export function constructRosettaError(errorCode: ApiError, details?: object) {
   const { message, retriable, code } = errors[errorCode];
-  return Error.constructFromObject({ code, message, retriable, details });
+  return RosettaClientError.constructFromObject({ code, message, retriable, details });
 }
 
-export function throwError(errorCode: ApiError, details?: object) {
-  throw constructRosettaError(errorCode, details);
+export function throwError(errorCode: ApiError, metadata?: object, error?: Error) {
+  let rosettaDetails: {[index: string]: any; } = {};
```

Comment:
```suggestion
  let rosettaDetails: Record<string, any> = {};
```

Reply:
@osipov-mit also updated in `logger.ts` 👍🏽 

https://github.com/gear-foundation/integrations-rosetta-api/pull/60/commits/038ae32e12a5bd582b50fdc6505f7c9fc61f220d